lib/sth_database.js

```diff
@@ -292,6 +316,115 @@ function getCollectionName4Aggregated(params) {
 }
 
 /**
+ * Sets the unique index for the raw data collections
+ * @param {object} collection The raw data collection
+ */
+function setRawDataUniqueIndex(collection) {
+  var uniqueCompoundIndex;
+  switch(sthConfig.DATA_MODEL) {
+    case sthConfig.DATA_MODELS.COLLECTION_PER_SERVICE_PATH:
+      uniqueCompoundIndex = {
+        'recvTime': 1,
+        'entityId': 1,
+        'entityType': 1,
+        'attrName': 1,
+        'attrType': 1,
+        'attrValue': 1
+      };
+      break;
+    case sthConfig.DATA_MODELS.COLLECTION_PER_ENTITY:
+      uniqueCompoundIndex = {
+        'recvTime': 1,
+        'attrName': 1,
+        'attrType': 1,
+        'attrValue': 1
+      };
+      break;
+    case sthConfig.DATA_MODELS.COLLECTION_PER_ATTRIBUTE:
+      uniqueCompoundIndex = {
+        'recvTime': 1,
+        'attrType': 1,
+        'attrValue': 1
+      };
+      break;
+  }
+  collection.ensureIndex(
+    uniqueCompoundIndex,
+    { unique: true },
+    function (err) {
+      if (err) {
+        sthLogger.error(
+          sthConfig.LOGGING_CONTEXT.DB_LOG,
+          'Error when creating the unique compound index for collection \'' +
+          collection.s.name + '\': ' + err
+        );
+      }
+    }
+  );
+}
+
+/**
+ * Returns true is the collection name corresponds to an aggregated data collection. False otherwise.
+ * @param collectionName The collection name
+ * @return {boolean} True is the collection name corresponds to an aggregated data collection. False otherwise.
+ */
+function isAggregated(collectionName) {
+  if (collectionName.lastIndexOf('.aggr') === -1 ||
+    collectionName.lastIndexOf('.aggr') !== collectionName.length - '.aggr'.length) {
+    return false;
+  } else {
+    return true;
+  }
+}
+
+/**
+ * Sets the time to live policy on a collection
+ * @param {object} collection The collection
+ */
+function setTTLPolicy(collection) {
+  // Set the TTL policy if required
+  if (sthConfig.TRUNCATION_EXPIREAFTERSECONDS > 0) {
+    if (!isAggregated(collection)) {
+      if (sthConfig.TRUNCATION_SIZE === 0) {
+        collection.ensureIndex(
+          {
+            'recvTime': 1
+          },
+          {
+            expireAfterSeconds: sthConfig.TRUNCATION_EXPIREAFTERSECONDS
+          }, function (err) {
+            if (err) {
+              sthLogger.error(
+                sthConfig.LOGGING_CONTEXT.DB_LOG,
+                'Error when creating the index for TTL for collection \'' +
+                collection.s.name + '\': ' + err
+              );
+            }
+          }
+        );
+      }
+    } else {
+      collection.ensureIndex(
+        {
+          '_id.origin': 1
+        },
+        {
+          expireAfterSeconds: sthConfig.TRUNCATION_EXPIREAFTERSECONDS
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-sth-comet/pull/332/commits/db8062f34c184fe174176a320019af3eedd6f7b1